rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -223,20 +221,18 @@ __check_parameter_value_in_range(
   }
 
   if (!descriptor.floating_point_range.empty() && value.get_type() == rclcpp::PARAMETER_DOUBLE) {
-    double v = value.get<double>();
-    auto fp_range = descriptor.floating_point_range.at(0);
-    if (__are_doubles_equal(v, fp_range.from_value) || __are_doubles_equal(v, fp_range.to_value)) {
-      return result;
-    }
-    if ((v < fp_range.from_value) || (v > fp_range.to_value)) {
+    const double v = value.get<double>();
+    const auto& fp_range = descriptor.floating_point_range.at(0);
+    const bool within_range = (v >= fp_range.from_value) && (v <= fp_range.to_value);
```

Comment:
This removed the epsilon, shouldn't it be 
```suggestion
    const bool within_range = (v >= fp_range.from_value - std::numeric_limits<double>::epsilon() ) && (v <= fp_range.to_value +  + std::numeric_limits<double>::epsilon() );
```

Also, if the __are_doubles_equal function is obviously broken, why was it not removed ?